### ISSUE TITLE
The documentation for Nan::New is inconsistent

### ISSUE THREAD

**Issue by murgatroid99** [nodejs/nan#466]
It says

> The following types will be contained within a `Nan::MaybeLocal<T>`: `v8::String`, `v8::Date`, `v8::RegExp`, `v8::Script`, `v8::UnboundScript`.

but the signatures given are

```cpp
v8::Local<v8::String> Nan::New<T>(std::string const& value);
v8::Local<v8::String> Nan::New<T>(const char * value, int length);
v8::Local<v8::String> Nan::New<T>(const char * value);
v8::Local<v8::String> Nan::New<T>(const uint16_t * value);
```

**Comment by kkoopa** [comment]
Will fix.

**Comment by brodybits** [comment]
I am not so sure that the documentation is wrong. Looking at the following sections:

[nan_new.h#L42-L44](./nan_new.h#L42-L44):
```c++
template <typename T> struct MaybeFactoryBase {
  typedef MaybeLocal<T> return_t;
};
```
[nan_new.h#L158-L172](./nan_new.h#L158-L172):
```c++
template <>
struct Factory<v8::String> : MaybeFactoryBase<v8::String> {
  static inline return_t New();
  static inline return_t New(const char *value, int length = -1);
  static inline return_t New(const uint16_t *value, int length = -1);
  static inline return_t New(std::string const& value);

  static inline return_t New(v8::String::ExternalStringResource * value);
  static inline return_t New(ExternalOneByteStringResource * value);
};
```
and [nan_new.h#L290-L300](./nan_new.h#L290-L300) (and further):
```c++
inline
imp::Factory<v8::String>::return_t
New(std::string const& value) {  // NOLINT(build/include_what_you_use)
  return New<v8::String>(value);
}

inline
imp::Factory<v8::String>::return_t
New(const char * value, int length) {
  return New<v8::String>(value, length);
}
```

I would conclude that the `v8::String` variants of `Nan::New<T>` also return `v8::MaybeLocal<T>` and *not* `v8::Local<T>`.

As an experiment, I tried building with `Factory<v8::String>` derived from `FactoryBase<v8::String>` instead of `MaybeFactoryBase<v8::String>` but encountered some errors since certain V8 functions are returning `MaybeLocal<v8::String>`.

**Comment by murgatroid99** [comment]
@brodybits The reason I opened this bug is that part of the documentation says that the `v8::String` variants return a `MaybeLocal` and another part says that they return a `Local`. No matter what the actual value is, one of those statements has to be wrong

**Comment by brodybits** [comment]
@murgatroid99 can you point out which part of the documentation says that the `v8::String` variants or functions return `v8::Local<String>`? I must admit that I am still pretty new to this project.

**Comment by murgatroid99** [comment]
The original issue quotes the code block that says that. It's in the page that describes `Nan::New`

**Comment by brodybits** [comment]
@murgatroid99 you are absolutely right! In the case of a string-like argument, `Nan::New` returns `Nan::MaybeLocal<v8::String>`. I will raise a fix for this.

**Comment by brodybits** [comment]
Raised PR #480 with the fix.

**Comment by brodybits** [comment]
Fix in #480 landed as 0a2af1e in `master` for 2.1. @murgatroid99 can you please close this if you are satisfied? Thanks.
